### PR TITLE
TEST-#6009: use `tmp_path` fixture instead of `ensure_clean_dir` as pandas 2.0.0 does

### LINKED ISSUE #6009: use `tmp_path` fixture instead of `ensure_clean_dir` as pandas 2.0.0 does

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

This is necessary step to migrate to pandas 2.0.0.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6009 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/storage_formats/pandas/parsers.py [hunk 1]
```diff
@@ -780,7 +781,7 @@ def parse(files_for_parser, engine, **kwargs):
         storage_options = kwargs.get("storage_options", {})
         chunks = []
         # `single_worker_read` just passes in a string path
-        if isinstance(files_for_parser, str):
+        if isinstance(files_for_parser, (str, Path)):
```
Comment (vnlitvinov) [inline modin/core/storage_formats/pandas/parsers.py:784]: ninja-fix of something else hidden in `TEST-` PR? [tut-tut](https://ru.wiktionary.org/wiki/tut-tut)...

if that is only needed for tests, one can call the parser like `pd.read_csv(str(path_obj))`

**Review by vnlitvinov — COMMENTED** [review]
I wonder whether Pytest is cleaning up its `tmp_path` after run... I can find nothing about that in the Pytest docs... 😞

**Comment by anmyachev** [comment]
> I wonder whether Pytest is cleaning up its `tmp_path` after run... I can find nothing about that in the Pytest docs... 😞

By default, after every 3rd time: https://docs.pytest.org/en/6.2.x/reference.html?highlight=tmp_path#std-fixture-tmp_path

File: modin/core/storage_formats/pandas/parsers.py (hunk 1 above)
Comment (anmyachev) [reply modin/core/storage_formats/pandas/parsers.py]: It's needed for parquet tests. Updated

Comment (anmyachev) [reply modin/core/storage_formats/pandas/parsers.py]: https://github.com/modin-project/modin/issues/6015

**Comment by anmyachev** [comment]
@vnlitvinov answered all your comments

**Comment by anmyachev** [comment]
> LGTM
> 
> @vnlitvinov any more comments?

@dchigarev let's merge it then

**Pull request merged**